index.html

```diff
@@ -876,28 +875,39 @@ <h3>
         <ol class="algorithm" data-link-for="WakeLock">
           <li>Let |type| be |lock|'s <a>type</a>.
           </li>
-          <li>If the promise |lock|.<a>[[\requestPromise]]</a> not `undefined`
-          and is pending, reject it with an "{{AbortError}}" {{DOMException}}
-          and <a>abort the request steps</a> for the associated running
-          <a>request()</a> algorithm.
+          <li>Let |record| be the <a>platform wake lock</a>'s <a>state
+          record</a> associated with |type|.
           </li>
-          <li>Ask the underlying operation system to release the wake lock of
-          type |type| and let |success| be `true` if the operation succeeded,
-          or else `false`.
+          <li>If the promise
+            |lock|.<a>[[\requestPromise]]</a> is not `undefined` and is
+            pending, reject it with an "{{AbortError}}" {{DOMException}}
```

Comment:
Checking for pending feels like it should reject with an InvalidStateError.

Reply:
This is doing abortion, so I think AbortError makes sense - abortion may very well be intentional... and even if the system aborts it, I would like to know... and not just get an invalid state